src/libfetchers/git-lfs-fetch.hh

```diff
@@ -0,0 +1,440 @@
+#include <array>
+#include <cstdlib>
+#include <curl/curl.h>
+#include <filesystem>
+#include <fstream>
+#include <git2.h>
+#include <iostream>
+#include <memory>
+#include <nlohmann/json.hpp>
+#include <regex>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+
+#include "serialise.hh"
+#include "processes.hh"
+#include "url.hh"
+
+namespace fs = std::filesystem;
+
+namespace nix {
+namespace lfs {
+
+// git-lfs metadata about a file
+struct Md
+{
+    std::string path; // fs path relative to repo root, no ./ prefix
+    std::string oid;  // git-lfs managed object id. you give this to the lfs server
+                      // for downloads
+    size_t size;      // in bytes
+};
+
+struct GitUrl
```

Comment:
This class looks very similar to `ParsedUrl`, maybe you can use that instead (in conjuction with `fixGitURL()` to handle the scp syntax)?